state/txindex/kv/kv.go

```diff
@@ -615,31 +608,38 @@ LOOP:
 
 // Keys
 
-func isTagKey(key []byte) bool {
-	return strings.Count(string(key), tagKeySeparator) == 3
-}
-
-func extractValueFromKey(key []byte) string {
-	parts := strings.SplitN(string(key), tagKeySeparator, 3)
-	return parts[1]
+func extractValueFromKey(key []byte) (value string) {
+	var prefix string
+	_, _ = orderedcode.Parse(string(key), &prefix, &value)
+	return
 }
 
 func keyForEvent(key string, value []byte, result *abci.TxResult) []byte {
-	return []byte(fmt.Sprintf("%s/%s/%d/%d",
+	encodedKey, err := orderedcode.Append(
+		nil,
 		key,
-		value,
+		string(value),
 		result.Height,
-		result.Index,
-	))
+		int64(result.Index),
+	)
+	if err != nil {
+		panic(err)
+	}
+	return encodedKey
 }
 
 func keyForHeight(result *abci.TxResult) []byte {
-	return []byte(fmt.Sprintf("%s/%d/%d/%d",
+	key, err := orderedcode.Append(
+		nil,
 		types.TxHeightKey,
 		result.Height,
 		result.Height,
```

Comment:
What's the reason for doing height twice? Is it so we still have the same format?

Reply:
Maybe, no idea. The query code might have hints.

Reply:
I think this is connected with my other question. We added the extra field here so that both styles of keys (`keyForHeight` and `keyForEvent`) had the same form .../.../.../....

Reply:
Ah, maybe I misunderstood your other question then -- we'll need to validate all external inputs in any case, but I don't have enough context to say whether the same or different key formats make sense.